manifests-gen/customizations.go

```diff
@@ -326,3 +334,59 @@ func mergeMaps[K comparable, V any](maps ...map[K]V) map[K]V {
 	}
 	return result
 }
+
+// addInfraClusterProtectionPolicy adds a Validating Admission Policy and Binding for protecting
+// InfraClusters created by the cluster-capi-operator from deletion and editing.
+func addInfraClusterProtectionPolicy(objs []unstructured.Unstructured, providerName string) []unstructured.Unstructured {
+	policy := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "admissionregistration.k8s.io/v1beta1",
+			"kind":       "ValidatingAdmissionPolicy",
+			"metadata": map[string]interface{}{
+				"name": "protect-capi-" + providerName + "cluster",
+			},
+			"spec": map[string]interface{}{
+				"failurePolicy": "Fail",
+				"matchConstraints": map[string]interface{}{
+					"resourceRules": []interface{}{
+						map[string]interface{}{
+							"apiGroups":   []interface{}{"infrastructure.cluster.x-k8s.io"},
+							"apiVersions": []interface{}{"*"},
+							"operations":  []interface{}{"UPDATE", "DELETE"},
+							"resources":   []interface{}{providerName + "clusters"},
+						},
+					},
+				},
+				"validations": []interface{}{
+					map[string]interface{}{
+						"expression": "!(oldObject.metadata.annotations['" + clusterv1.ManagedByAnnotation + "'] == '" + managedByAnnotationValueClusterCAPIOperatorInfraClusterController + "')",
+						"message":    "InfraCluster resources with the '" + clusterv1.ManagedByAnnotation + " : " + managedByAnnotationValueClusterCAPIOperatorInfraClusterController + "' annotation cannot be edited or deleted.",
+					},
+				},
+			},
+		},
+	}
+
+	binding := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "admissionregistration.k8s.io/v1beta1",
+			"kind":       "ValidatingAdmissionPolicyBinding",
+			"metadata": map[string]interface{}{
+				"name": "protect-capi-" + providerName + "cluster-binding",
```

Comment:
Naming as above, but we don't need the `binding` suffix IMO, we already have the resource type by the time we see the name

Reply:
Done